File: apps/backend/app/models/knowledge_base.py
from datetime import datetime
from enum import Enum
from typing import Any, Dict, Literal
from pydantic import BaseModel, Field
from bson import ObjectId
from .user import PyObjectId
from beanie import Document

class KnowledgeBaseType(str, Enum):
    PDF = "pdf"
    CSV = "csv"
    LINK = "link"
    VIDEO = "video"
    TEXT = "text"
    IMAGE = "image"
    AUDIO = "audio"
    XLSX = "xlsx"

class KnowledgeBase(Document):
    id: PyObjectId = Field(default_factory=PyObjectId, alias="_id")
    workspace_id: PyObjectId
    type: KnowledgeBaseType
    file_url: str
    created_at: datetime = Field(default_factory=datetime.now)
    updated_at: datetime = Field(default_factory=datetime.now)
    name: str
    status: Literal["pending", "indexed", "failed"] = "indexed"
    metadata: Dict[str, Any] = Field(default_factory=dict)

    class Config:
        json_encoders = {ObjectId: str}

    class Settings:
        name = "knowledge_bases"

File: apps/frontend/src/types/knowledge-base.ts
export type KnowledgeBaseType =
  | "pdf"
  | "csv"
  | "link"
  | "video"
  | "text"
  | "image"
  | "audio"
  | "xlsx"

export type KnowledgeBaseItem = {
  _id: string
  workspace_id: string
  type: KnowledgeBaseType
  file_url: string
  created_at: string
  updated_at: string
  name: string
  status?: "pending" | "indexed" | "failed"
  metadata?: Record<string, unknown>
}

export type CrawlRequest = {
  base_url: string
  max_pages: number
  max_depth: number
  include_paths: string[]
  exclude_paths: string[]
}

export type CrawlJob = {
  job_id: string
  workspace_id: string
  job_type: string
  status: "pending" | "running" | "completed" | "failed"
  message?: string | null
  processed_items: number
  total_items: number
  payload: Partial<CrawlRequest>
  created_at: string
  updated_at: string
  completed_at?: string | null
}
